### PR TITLE
AJJGenJet filter added.

### PR DESCRIPTION
**Pull request by hbakhshi**
#### PR description:
A new filter added to the GeneratorInterface/GenFilters package. It is specially designed to select events with 1 photon + 2VBF jets. cuts on photon and jet are completely configurable.

#### PR validation:

the filter tested on run-2 and snowmass gridpacks and fragments. 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

this filter is going to be used to generate vbf+photon samples for run2 and snowmass analyses

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: GeneratorInterface/GenFilters/plugins/AJJGenJetFilter.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#include <map>
+#include <vector>
+
+using namespace edm;
+using namespace std;
+//
+// class declaration
+//
+
+class AJJGenJetFilter : public edm::EDFilter {
```
Comment (Dr15Jones) [inline GeneratorInterface/GenFilters/plugins/AJJGenJetFilter.h:34]: Please use a thread friendly base class.

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by hbakhshi** [comment]
Should I do something special for the test? I have already tested it with my gridpacks and it works.

**Comment by Dr15Jones** [comment]
@cmsbuild please test

**Comment by Saptaparna** [comment]
@hbakhshi let's wait to hear from @SiewYan and @Dr15Jones. @SiewYan since this a VBF filter, should we test a VBF-like workflow?

**Comment by hbakhshi** [comment]
@SiewYan @Saptaparna thanks a lot for triggering the test. what should I expect now? passing the tests, can it be considered as merged?

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by SiewYan** [comment]
please test workflow 552,554

**Comment by SiewYan** [comment]
Hello @hbakhshi , The three workflows used for the test are GGToHgg_NLO_Pow_13TeV_py8 (514), VHToHtt_NLO_Pow_13TeV_py8 (552) and VBFToH4l_NLO_Pow_JHU_13TeV_py8 (554), they are defined in [1]. For all intents and purposes, these tests (VBF-like) are suffice to understand if those added filters compiled and worked, and they compiled and successful.

Unfortunately the baseline comparison is not available (due to missing matrix map), otherwise we may look at the set of DQM plots.

I suggest to approve this PR, @Saptaparna @GurpreetSinghChahal , what do you think?

[1] https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_extendedgen.py#L34-L36

**Comment by Saptaparna** [comment]
It looks good. Just one question:
```
minEta = cms.untracked.double( -4.5 ),
maxEta = cms.untracked.double( 4.5 ),
```
Why not extend it to 5.0?

**Comment by hbakhshi** [comment]
Looking into some delphes files for HL-LHC, I guessed the CMS endcap goes up to 4.5 
If I am wrong and it extends to 5, I will change it. Please let me know.

**Comment by hbakhshi** [comment]
It seems that all the tests have passed again. I would appreciate if you could accept this PR, so I can proceed with the sample request. @Saptaparna

**Pull request merged**